### ISSUE TITLE
RuntimeError: Could not run 'torchvision::nms' with arguments from the 'XPU' backend.

### ISSUE THREAD

**Issue by xingjian-zhang** [intel/intel-extension-for-pytorch#167]
```python
# Sample code from the TorchVision 0.3 Object Detection Finetuning Tutorial
# http://pytorch.org/tutorials/intermediate/torchvision_tutorial.html

import os
import math
import sys
import time
import numpy as np
import torch
import argparse
from PIL import Image

import torchvision
from torchvision.models.detection.faster_rcnn import FastRCNNPredictor
from torchvision.models.detection.mask_rcnn import MaskRCNNPredictor

from coco_utils import get_coco_api_from_dataset
import utils
from coco_eval import CocoEvaluator
import transforms as T
import pytorch_lightning as pl

from orca_lite.accelerator.ipex_accelerator import IPEXAccelerator
from orca_lite import init_orca_lite

parser = argparse.ArgumentParser(description='PyTorch Penn-Fudan')
parser.add_argument('--ipex', action='store_true', default=False,
                    help='use intel pytorch extension')

parser.add_argument('--batch_size', default=2, type=int, help='batch size')
parser.add_argument('--epochs', default=1, type=int, help='epochs to train')
parser.add_argument('--data_path', default='../../data/PennFudanPed',
                    help='path to cat vs dog dataset')
parser.add_argument('-j', '--jemalloc', action='store_false', default=True,
                    help='whether to use jemalloc')
parser.add_argument('-o', '--openmp', action='store_false', default=True,
                    help='whether to use openmp')


class PennFudanDataset(object):
    def __init__(self, root, transforms):
        self.root = root
        self.transforms = transforms
        # load all image files, sorting them to
        # ensure that they are aligned
        self.imgs = list(sorted(os.listdir(os.path.join(root, "PNGImages"))))
        self.masks = list(sorted(os.listdir(os.path.join(root, "PedMasks"))))

    def __getitem__(self, idx):
        # load images and masks
        img_path = os.path.join(self.root, "PNGImages", self.imgs[idx])
        mask_path = os.path.join(self.root, "PedMasks", self.masks[idx])
        img = Image.open(img_path).convert("RGB")
        # note that we haven't converted the mask to RGB,
        # because each color corresponds to a different instance
        # with 0 being background
        mask = Image.open(mask_path)

        mask = np.array(mask)
        # instances are encoded as different colors
        obj_ids = np.unique(mask)
        # first id is the background, so remove it
        obj_ids = obj_ids[1:]

        # split the color-encoded mask into a set
        # of binary masks
        masks = mask == obj_ids[:, None, None]

        # get bounding box coordinates for each mask
        num_objs = len(obj_ids)
        boxes = []
        for i in range(num_objs):
            pos = np.where(masks[i])
            xmin = np.min(pos[1])
            xmax = np.max(pos[1])
            ymin = np.min(pos[0])
            ymax = np.max(pos[0])
            boxes.append([xmin, ymin, xmax, ymax])

        boxes = torch.as_tensor(boxes, dtype=torch.float32)
        # there is only one class
        labels = torch.ones((num_objs,), dtype=torch.int64)
        masks = torch.as_tensor(masks, dtype=torch.uint8)

        image_id = torch.tensor([idx])
        area = (boxes[:, 3] - boxes[:, 1]) * (boxes[:, 2] - boxes[:, 0])
        # suppose all instances are not crowd
        iscrowd = torch.zeros((num_objs,), dtype=torch.int64)

        target = {}
        target["boxes"] = boxes
        target["labels"] = labels
        target["masks"] = masks
        target["image_id"] = image_id
        target["area"] = area
        target["iscrowd"] = iscrowd

        if self.transforms is not None:
            img, target = self.transforms(img, target)

        return img, target

    def __len__(self):
        return len(self.imgs)


def get_model_instance_segmentation(num_classes):
    # load an instance segmentation model pre-trained pre-trained on COCO
    model = torchvision.models.detection.maskrcnn_resnet50_fpn(pretrained=True)

    # get number of input features for the classifier
    in_features = model.roi_heads.box_predictor.cls_score.in_features
    # replace the pre-trained head with a new one
    model.roi_heads.box_predictor = FastRCNNPredictor(in_features, num_classes)

    # now get the number of input features for the mask classifier
    in_features_mask = model.roi_heads.mask_predictor.conv5_mask.in_channels
    hidden_layer = 256
    # and replace the mask predictor with a new one
    model.roi_heads.mask_predictor = MaskRCNNPredictor(in_features_mask,
                                                       hidden_layer,
                                                       num_classes)

    return model


def _get_iou_types(model):
    model_without_ddp = model
    if isinstance(model, torch.nn.parallel.DistributedDataParallel):
        model_without_ddp = model.module
    iou_types = ["bbox"]
    if isinstance(model_without_ddp, torchvision.models.detection.MaskRCNN):
        iou_types.append("segm")
    if isinstance(model_without_ddp, torchvision.models.detection.KeypointRCNN):
        iou_types.append("keypoints")
    return iou_types


class Net(pl.LightningModule):
    def __init__(self, num_classes, coco_api, len_dataloader):
        super().__init__()
        self.model = get_model_instance_segmentation(num_classes).train()
        self.optimizer_params = [p for p in self.model.parameters() if p.requires_grad]
        self.evaluator = CocoEvaluator(coco_api, _get_iou_types(self.model))
        self.len_dataloader = len_dataloader

    def forward(self, x):
        return self.model(x)

    def training_step(self, batch, batch_idx):
        x, y = batch
        loss_dict = self.model(x, y)
        losses = sum(loss for loss in loss_dict.values())
        # reduce losses over all GPUs for logging purposes
        loss_dict_reduced = utils.reduce_dict(loss_dict)
        losses_reduced = sum(loss for loss in loss_dict_reduced.values())

        loss_value = losses_reduced.item()

        if not math.isfinite(loss_value):
            print("Loss is {}, stopping training".format(loss_value))
            print(loss_dict_reduced)
            sys.exit(1)
        self.log('train_loss', loss_value)

        return losses

    def test_step(self, batch, batch_idx):
        x, y = batch
        outputs = self.model(x)
        outputs = [{k: v for k, v in t.items()} for t in outputs]
        res = {target["image_id"].item(): output for target, output in zip(y, outputs)}

        self.evaluator.update(res)

    def on_test_end(self) -> None:
        self.evaluator.synchronize_between_processes()

        # accumulate predictions from all images
        self.evaluator.accumulate()
        self.evaluator.summarize()
        return super().on_test_end()

    def configure_optimizers(self):
        optimizer = torch.optim.SGD(self.optimizer_params, lr=0.005,
                                    momentum=0.9, weight_decay=0.0005)
        lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer,
                                                       step_size=3,
                                                       gamma=0.1)
        return ({
            'optimizer': optimizer,
            'lr_scheduler': {
                'scheduler': lr_scheduler,
                'interval': 'epoch',
                'frequency': 1
            }
        })


def get_transform(train):
    transforms = []
    transforms.append(T.ToTensor())
    if train:
        transforms.append(T.RandomHorizontalFlip(0.5))
    return T.Compose(transforms)


def main():
    args = parser.parse_args()
    init_orca_lite(args.jemalloc, args.openmp)

    # our dataset has two classes only - background and person
    num_classes = 2
    # use our dataset and defined transformations
    dataset_root = args.data_path
    dataset = PennFudanDataset(dataset_root, get_transform(train=True))
    dataset_test = PennFudanDataset(dataset_root, get_transform(train=False))

    # split the dataset in train and test set
    indices = torch.randperm(len(dataset)).tolist()
    dataset = torch.utils.data.Subset(dataset, indices[:-50])
    dataset_test = torch.utils.data.Subset(dataset_test, indices[-50:])

    # define training and validation data loaders
    data_loader = torch.utils.data.DataLoader(
        dataset, batch_size=args.batch_size, shuffle=True, num_workers=0,
        collate_fn=utils.collate_fn)

    data_loader_test = torch.utils.data.DataLoader(
        dataset_test, batch_size=1, shuffle=False, num_workers=0,
        collate_fn=utils.collate_fn)

    # let's train it for 1 epochs
    num_epochs = args.epochs

    coco = get_coco_api_from_dataset(data_loader_test.dataset)
    net = Net(num_classes=num_classes, coco_api=coco,
              len_dataloader=len(data_loader))

    if args.ipex:
        trainer = pl.Trainer(max_epochs=num_epochs, accelerator=IPEXAccelerator())
    else:
        trainer = pl.Trainer(max_epochs=num_epochs)

    start_time = time.time()
    trainer.fit(net, data_loader)
    end_time = time.time()
    throughput = len(data_loader) / (end_time - start_time)

    print("Result:")
    print("average throughput {}".format(throughput))
    trainer.test(net, data_loader_test)

    print("That's it!")


if __name__ == "__main__":
    main()
```
When I run the above code with ipex enabled, the following error occurs:
```
Traceback (most recent call last):
  File "tv-training-code.py", line 258, in <module>
    main()
  File "tv-training-code.py", line 246, in main
    trainer.fit(net, data_loader)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 458, in fit
    self._run(model)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 756, in _run
    self.dispatch()
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 797, in dispatch
    self.accelerator.start_training(self)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/accelerators/accelerator.py", line 96, in start_training
    self.training_type_plugin.start_training(trainer)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/plugins/training_type/training_type_plugin.py", line 144, in start_training
    self._results = trainer.run_stage()
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 807, in run_stage
    return self.run_train()
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 869, in run_train
    self.train_loop.run_training_epoch()
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/trainer/training_loop.py", line 489, in run_training_epoch
    batch_output = self.run_training_batch(batch, batch_idx, dataloader_idx)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/trainer/training_loop.py", line 728, in run_training_batch
    self.optimizer_step(optimizer, opt_idx, batch_idx, train_step_and_backward_closure)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/trainer/training_loop.py", line 432, in optimizer_step
    using_lbfgs=is_lbfgs,
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/core/lightning.py", line 1403, in optimizer_step
    optimizer.step(closure=optimizer_closure)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/core/optimizer.py", line 214, in step
    self.__optimizer_step(*args, closure=closure, profiler_name=profiler_name, **kwargs)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/core/optimizer.py", line 134, in __optimizer_step
    trainer.accelerator.optimizer_step(optimizer, self._optimizer_idx, lambda_closure=closure, **kwargs)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/accelerators/accelerator.py", line 329, in optimizer_step
    self.run_optimizer_step(optimizer, opt_idx, lambda_closure, **kwargs)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/accelerators/accelerator.py", line 336, in run_optimizer_step
    self.training_type_plugin.optimizer_step(optimizer, lambda_closure=lambda_closure, **kwargs)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/plugins/training_type/training_type_plugin.py", line 193, in optimizer_step
    optimizer.step(closure=lambda_closure, **kwargs)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/torch/optim/lr_scheduler.py", line 65, in wrapper
    return wrapped(*args, **kwargs)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/torch/optim/optimizer.py", line 89, in wrapper
    return func(*args, **kwargs)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/torch/optim/sgd.py", line 87, in step
    loss = closure()
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/trainer/training_loop.py", line 723, in train_step_and_backward_closure
    split_batch, batch_idx, opt_idx, optimizer, self.trainer.hiddens
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/trainer/training_loop.py", line 813, in training_step_and_backward
    result = self.training_step(split_batch, batch_idx, opt_idx, hiddens)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/trainer/training_loop.py", line 280, in training_step
    training_step_output = self.trainer.accelerator.training_step(args)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/accelerators/accelerator.py", line 204, in training_step
    return self.training_type_plugin.training_step(*args)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/pytorch_lightning/plugins/training_type/training_type_plugin.py", line 155, in training_step
    return self.lightning_module.training_step(*args, **kwargs)
  File "tv-training-code.py", line 152, in training_step
    loss_dict = self.model(x, y)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/torchvision/models/detection/generalized_rcnn.py", line 97, in forward
    proposals, proposal_losses = self.rpn(images, features, targets)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/torchvision/models/detection/rpn.py", line 357, in forward
    boxes, scores = self.filter_proposals(proposals, objectness, images.image_sizes, num_anchors_per_level)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/torchvision/models/detection/rpn.py", line 272, in filter_proposals
    keep = box_ops.batched_nms(boxes, scores, lvl, self.nms_thresh)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/torch/jit/_trace.py", line 1091, in wrapper
    return fn(*args, **kwargs)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/torchvision/ops/boxes.py", line 75, in batched_nms
    keep = nms(boxes_for_nms, scores, iou_threshold)
  File "/opt/conda/envs/orca_lite/lib/python3.7/site-packages/torchvision/ops/boxes.py", line 36, in nms
    return torch.ops.torchvision.nms(boxes, scores, iou_threshold)
RuntimeError: Could not run 'torchvision::nms' with arguments from the 'XPU' backend. This could be because the operator doesn't exist for this backend, or was omitted during the selective/custom build process (if using custom build). If you are a Facebook employee using PyTorch on mobile, please visit https://fburl.com/ptmfixes for possible resolutions. 'torchvision::nms' is only available for these backends: [CPU, BackendSelect, Named, AutogradOther, AutogradCPU, AutogradCUDA, AutogradXLA, Tracer, Autocast, Batched, VmapMode].

CPU: registered at /root/project/torchvision/csrc/ops/cpu/nms_kernel.cpp:111 [kernel]
BackendSelect: fallthrough registered at /pytorch/aten/src/ATen/core/BackendSelectFallbackKernel.cpp:3 [backend fallback]
Named: registered at /pytorch/aten/src/ATen/core/NamedRegistrations.cpp:7 [backend fallback]
AutogradOther: fallthrough registered at /pytorch/aten/src/ATen/core/VariableFallbackKernel.cpp:35 [backend fallback]
AutogradCPU: fallthrough registered at /pytorch/aten/src/ATen/core/VariableFallbackKernel.cpp:39 [backend fallback]
AutogradCUDA: fallthrough registered at /pytorch/aten/src/ATen/core/VariableFallbackKernel.cpp:43 [backend fallback]
AutogradXLA: fallthrough registered at /pytorch/aten/src/ATen/core/VariableFallbackKernel.cpp:47 [backend fallback]
Tracer: fallthrough registered at /pytorch/torch/csrc/jit/frontend/tracer.cpp:999 [backend fallback]
Autocast: fallthrough registered at /pytorch/aten/src/ATen/autocast_mode.cpp:250 [backend fallback]
Batched: registered at /pytorch/aten/src/ATen/BatchingRegistrations.cpp:1016 [backend fallback]
VmapMode: fallthrough registered at /pytorch/aten/src/ATen/VmapModeRegistrations.cpp:33 [backend fallback]

```

**Comment by roshnikasliwal** [comment]
HI xingjian-zhang,
Can you tell me how you resolved this?

**Comment by junjihashimoto** [comment]
This issue is not solved.
I've tried `torchvision::nms` with ipex-v1.9.0.
Some modes like yolov5 and torchvision-fasterrcnn uses `torchvision::nms` in their models.
It would be helpful if you provides `torchvision::nms` for 'XPU' backend.

**Comment by EikanWang** [comment]
@junjihashimoto , regarding the coming ipex 1.10, we switched the underhood device back to `CPU`. From IPEX 1.10, we changed to the operator register mechanism. I'd recommend you to switch to IPEX 1.10.

**Comment by EikanWang** [comment]
https://github.com/intel/intel-extension-for-pytorch/tree/v1.10.0+cpu-rc3

**Comment by junjihashimoto** [comment]
@EikanWang 
Thx! I'll try the new one.

**Comment by EikanWang** [comment]
@junjihashimoto , we just released the 1.10.0. You can directly install ipex 1.10 binary now. https://intel.github.io/intel-extension-for-pytorch/tutorials/installation.html#

**Comment by JustSomeRandomUsername** [comment]
I seem to be having the same error:
`import torch
import torchvision
import intel_extension_for_pytorch as ipex

a = torch.zeros(1, device="xpu")
torch.ops.torchvision.nms(a,a,1)`

Causes "Could not run 'torchvision::nms' with arguments from the 'XPU' backend" error.

I am using a Arc770 on Ubuntu 22.04
python 3.9
ipex-version: 1.13.10+xpu
torchvision-version: 0.14.1a0+0504df5
